### PR TITLE
Migrate `LOCKING_*_ENABLE` to Data-Driven: I-J

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Moves instances of `#define LOCKING_SUPPORT_ENABLE` and `#define LOCKING_RESYNC_ENABLE` to data-driven.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

File: keyboards/jukaie/jk01/keyboard.json [hunk 1]
```diff
@@ -29,6 +29,12 @@
         "nkro": true,
         "rgb_matrix": true
     },
+    "qmk": {
```
Comment (zvecr) [inline keyboards/jukaie/jk01/keyboard.json:32]: duplicate key: 'qmk'

File: keyboards/joshajohnson/hub20/keyboard.json [hunk 2]
```diff
@@ -40,6 +40,12 @@
         "nkro": true,
         "rgb_matrix": true
     },
+    "qmk": {
```
Comment (zvecr) [inline keyboards/joshajohnson/hub20/keyboard.json:43]: duplicate key: 'qmk'

File: keyboards/inland/kb83/keyboard.json [hunk 3]
```diff
@@ -43,6 +43,12 @@
         "dip_switch": true,
         "encoder": true
     },
+    "qmk": {
```
Comment (zvecr) [inline keyboards/inland/kb83/keyboard.json:46]: duplicate key: 'qmk'

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/jukaie/jk01/keyboard.json (hunk 1 above)
Comment (noroadsleft) [reply keyboards/jukaie/jk01/keyboard.json]: [88dcbf6dce..204c12474a](https://github.com/qmk/qmk_firmware/compare/88dcbf6dce463b6af04909d51693869088e49714..204c12474a8b02c312fe34b23d48ac969cbbdebc)

**Review by zvecr — APPROVED** [review]

**Pull request merged**